src/coreclr/jit/lsraxarch.cpp

```diff
@@ -2304,12 +2304,17 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
 
                 if (!op2->OperIsConst() && !op1->isContained())
                 {
-                    // If the index is not a constant or op1 is in register,
+                    // If the index is not a constant and op1 is in register,
                     // we will use the SIMD temp location to store the vector.
 
                     var_types requiredSimdTempType = Compiler::getSIMDTypeForSize(intrinsicTree->GetSimdSize());
                     compiler->getSIMDInitTempVarNum(requiredSimdTempType);
                 }
+                else if (op1->IsCnsVec())
```

Comment:
Can we add the WithElement path at the same time, since we're trying to keep the optimizations between the two in sync?

Reply:
We can't do it there, because it has to get spilled for the fallback implementation.  Can't write to the data section.

Reply:
> re: matching this optimization for WithElement, that doesn't work because that one spills to the stack for writing -- this only helps when we can read from the data section instead of the spill.

Should've finished reading the comments first 😄